data/scenarios/Tutorials/cond.yaml

```diff
@@ -0,0 +1,112 @@
+name: Conditionals
+description: |
+  Learn how to write conditional expressions.
+objectives:
+  - goal:
+      - |
+        The `if` command can be used to write conditional expressions.  `if` is
+        not actually special syntax; it is simply a built-in function of type
+      - |
+        if : bool -> {a} -> {a} -> a.
+      - |
+        It takes a boolean expression and then returns either the first or second subsequent
+        argument, depending on whether the boolean expression is true or false, respectively.
+      - |
+        The `a` can stand for any type; `{a}` indicates a *delayed* expression of type
+        `a`.  Normally, function arguments are evaluated strictly before the function is
+        called. Delayed expressions, on the other hand, are not evaluated until needed.
+        In this case, we want to make sure that only the correct branch is evaluated.
+        To write a value of type, say, `{int}`, we just surround a value of type `int`
+        in curly braces, like `{3}`.  This is why arguments to `build` must also be in
+        curly braces: the type of `build` is `{cmd a} -> cmd robot`.
+      - |
+        The 8x8 gray square contains 8 `very small rock`s --- so small they cannot be seen!
+        Your goal is to collect all of them and bring them back to your base; you win
+        when you have all 8.  They are not located in any particular pattern, so your best
+        bet is to sweep over the entire 8x8 square and pick up a `very small rock` any time
+        you detect one.
+      - |
+        TIP: You may find `ishere : string -> cmd bool` helpful for detecting rocks.
+      - |
+        TIP: Note that `if` requires a `bool`, not a `cmd bool`!  So you cannot directly say
+        `if (ishere "very small rock") {...} {...}`.  Instead you can write `b <- ishere "very small rock"; if b {...} {...}`.  You might enjoy writing your own function of
+        type `cmd bool -> {cmd a} -> {cmd a} -> cmd a` to encapsulate this pattern.
```

Comment:
Since the player has recently completed the lambda tutorial, this is a good chance to practice.  I'd rather give them a bit less guidance here.  (Though now that I think of it, the idea of having multi-argument curried functions with types like `a -> b -> c -> d` defined by `\x. \y. \z. ...` is not at all an obvious idea if you've never seen it before.  So that probably needs to be addressed somehow (but not here).  Perhaps eventually we'll have multiple tutorials, one for people who are already familiar with functional programming and one for those who aren't...)

In any case, the other annoying thing about this is that everything is automatically wrapped, so this code template looks bad since it wraps around the end.  Eventually, with #574 I assume code blocks will not be reformatted.